test/terraform_plan_details_example_test.go

```diff
@@ -0,0 +1,46 @@
+package test
+
+import (
+	"testing"
+
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	"github.com/stretchr/testify/assert"
+)
+
+// An example of how to test information in a terraform plan.  This tests that
+// a resource's name is correctly created from a value in a `for_each`
+// expression.
+func TestTerraformPlanDetailsExample(t *testing.T) {
+	outputPath := "/tmp/terraform-plan.plan"
+	terraformOptions := &terraform.Options{
+		TerraformDir: "../examples/terraform-plan-details-example",
+
+		// This will write a binary plan output file when plan is run
+		PlanOutputFile: outputPath,
+	}
+
+	terraform.InitAndPlan(t, terraformOptions)
+
+	// Outputs a JSON-formatted version of the terraform plan file
+	output := terraform.Show(t, terraformOptions, terraform.JSON, outputPath)
+
+	// Constructs a PlanInfo datastructure with information about the plan
+	info, err := terraform.NewPlanInfo(output)
+	assert.Nil(t, err)
```

Comment:
We like to use `require.NoError` when checking errors. The difference between `require` and `assert` is that `require` will fail the test and _end execution_ when the condition fails, where as `assert` only marks the test as failed but will continue execution. This can lead to test panic where you might reference a nil pointer.

Additionally, the difference between `Nil` and `NoError` is that `NoError` will show the error details when there is an error.